cloud/ec2_fleet.go

```diff
@@ -475,7 +514,13 @@ func (m *ec2FleetManager) makeOverrides(ctx context.Context, ec2Settings *EC2Pro
 	if len(subnets) > 0 {
 		overrides := make([]*ec2.FleetLaunchTemplateOverridesRequest, 0, len(subnets))
 		for _, subnet := range subnets {
-			overrides = append(overrides, &ec2.FleetLaunchTemplateOverridesRequest{SubnetId: aws.String(subnet.SubnetID)})
+			supported, err := typeCache.azSupportsInstanceType(ctx, m.client, subnet.AZ, ec2Settings.InstanceType)
```

Comment:
So are you caching because the same availability zone could be called multiple times here? Or because makeOverrides is called often?

Reply:
the latter.
`makeOverrides` is called once for each instance we start with Fleet, so instead of adding six API calls for each instance (one for each AZ) each app server will make six calls each time Evergreen is restarted.